packages/storage/src/public-types.ts

```diff
@@ -17,13 +17,10 @@
 
 // eslint-disable-next-line import/no-extraneous-dependencies
 import { FirebaseApp, _FirebaseService } from '@firebase/app';
-import {
-  CompleteFn,
-  FirebaseError,
-  NextFn,
-  Subscribe,
-  Unsubscribe
-} from '@firebase/util';
+import { CompleteFn, NextFn, Subscribe, Unsubscribe } from '@firebase/util';
+import { StorageError } from './implementation/error';
+
+export { StorageError, StorageErrorCode } from './implementation/error';
```

Comment:
I think this should be exported directly from `src/api.ts`. I think we had some weird bugs before when non-types were exported from a file that otherwise only exported types, but even if that's not a problem, just for organizational purposes.